app/main/forms.py

```diff
@@ -19,10 +21,34 @@ class ChangeEmailForm(FlaskForm):
         Required(), Email()])
 
 
+class RequiredIf(InputRequired):
+
+    def __init__(self, field_name, value, *args, **kwargs):
+        self.requiring_field_name = field_name
+        self.requiring_value = value
+        super(RequiredIf, self).__init__(*args, **kwargs)
+
+    def __call__(self, form, field):
+        requiring_field = form._fields.get(self.requiring_field_name)
+
+        if requiring_field:
+
+            if requiring_field.data == self.requiring_value:
+                super(RequiredIf, self).__call__(form, field)
+
+            else:
+                return Optional()(form, field)
+
+
 class EmailForm(FlaskForm):
     email_known = RadioField(choices=yesno)
     email_address = EmailField('Email', validators=[
```

Comment:
Would it be better if validators the parameters were moved to a new line to make the code clearer?